apps/frontend/src/store/data_filters.ts

```diff
@@ -353,18 +374,186 @@ export const FilteredDataModule = getModule(FilteredData);
  * - defaulting "omit_overlayed_controls"
  */
 export function filter_cache_key(f: Filter) {
-  // fix the search term
-  let newSearch: string;
-  if (f.searchTerm !== undefined) {
-    newSearch = f.searchTerm.trim();
-  } else {
-    newSearch = '';
-  }
-
   const newFilter: Filter = {
-    searchTerm: newSearch,
+    searchTerm: f.searchTerm?.trim() || '',
     omit_overlayed_controls: f.omit_overlayed_controls || false,
     ...f
   };
   return JSON.stringify(newFilter);
 }
+
+export function lowercaseAll(input: string | string[]): string | string[] {
+  if (typeof input === 'string') {
+    return input.toLowerCase();
+  } else {
+    return input.map((string) => {
+      return string.toLowerCase();
+    });
+  }
+}
+
+export function filterByStatus(
+  filter: Filter,
+  controls: readonly context.ContextualizedControl[]
+): context.ContextualizedControl[] {
+  if (filter.status && filter.status?.length !== 0) {
+    const foundControls: context.ContextualizedControl[] = [];
+    filter.status?.forEach(async (statusFilter) => {
+      if (statusFilter.toLowerCase() === 'waived') {
+        foundControls.push(...controls.filter((control) => control.hdf.waived));
+      } else {
+        foundControls.push(
+          ...controls.filter(
+            (control) =>
+              control.root.hdf.status.toLowerCase() ===
+              statusFilter.toLowerCase()
+          )
+        );
+      }
+    });
+    return foundControls.filter((c, index) => {
+      return foundControls.indexOf(c) === index;
+    });
+  } else {
+    return [...controls];
```

Comment:
This was originally done to avoid the read-only type of the original `controls` array, however, typing these filters as returning readonly also works.